WebSharper.Owin/Owin.Sitelets.fs

```diff
@@ -19,9 +18,45 @@ module P = WebSharper.PathConventions
 module M = WebSharper.Core.Metadata
 
 type Env = IDictionary<string, obj>
+type HeaderDictionary = IDictionary<string, string[]>
 type AppFunc = Func<Env, Task>
 type MidFunc = Func<AppFunc, AppFunc>
 
+// TODO: Use Arachne or Freya.Core instead.
+module Environment =

```

Comment:
Most of this should already have well defined parsers in Arachne, including the query string and Cookie support I still need to add.